data_structures/linked_list/circular_doubly_linked_list.c

```diff
@@ -0,0 +1,225 @@
+    uint64_t value = 0;                   ///< Data stored on each node
+};
+
+/**
+ * @brief   Insert the node at start of list
+ * @param   start   starting pointer of list
+ * @param   data    the data which we want to insert into list
+ * @returns void
+ */
+void insert_at_start(struct node **start, uint64_t data)
+{
+    struct node *n1;
+    n1 = (struct node *)malloc(sizeof(struct node));
+    n1->value = data;
+    if (*start == NULL)
+    {
+        n1->next = n1;
+        n1->prev = n1;
+        *start = n1;
+    }
+    else
+    {
+        struct node *temp;
+        temp = (*start)->prev;
+        n1->next = *start;
+        (*start)->prev = n1;
+        n1->prev = temp;
+        temp->next = n1;
+        *start = n1;
+    }
+    printf("\nNode succesfully inserted\n\n");
+}
+
+/**
+ * @brief   Insert the node at end of list
+ * @param   start   starting pointer of list
+ * @param   data    the data which we want to insert into list
+ * @returns void
+ */
+void insert_at_last(struct node **start, uint64_t data)
+{
+    struct node *n1;
+    n1 = (struct node *)malloc(sizeof(struct node));
+    n1->value = data;
+    if (*start == NULL)
+    {
+        n1->next = n1;
+        n1->prev = n1;
+        *start = n1;
+    }
+    else
+    {
+        struct node *temp1, *temp2;
+        temp1 = *start;
+        temp2 = (*start)->prev;
+        n1->prev = temp2;
+        n1->next = temp1;
+        temp1->prev = n1;
+        temp2->next = n1;
+    }
+    printf("\nNode succesfully inserted\n\n");
+}
+
+/**
+ * @brief   Deletion of the first node in list function
+ * @param   start   starting pointer of list
+ * @returns void
+ */
+void delete_first_node(struct node **start)
+{
+    if (*start == NULL)
+    {
+        printf("⚠️list is Empty⚠️\n");
+    }
+    else
+    {
+        struct node *temp1, *temp2;
+        temp1 = *start;
+        temp2 = temp1->prev;
+        if (temp1 == temp2)
+        {
+            free(temp2);
+            *start = NULL;
+        }
+        else
+        {
+            temp2->next = temp1->next;
+            (temp1->next)->prev = temp2;
+            *start = temp1->next;
+            free(temp1);
+            temp1 = NULL;
+        }
+        printf("\nNode succesfully deleted\n\n");
+    }
+}
+
+/**
+ * @brief   Deletion of the last node in list function
+ * @param   start   starting pointer of list
+ * @returns void
+ */
+void delete_last_node(struct node **start)
+{
+    if (*start == NULL)
+    {
+        printf("list is Empty\n");
+    }
+    else
+    {
+        struct node *temp1, *temp2;
+        temp1 = *start;
+        temp2 = temp1->prev;
+        if (temp1 == temp2)
+        {
+            free(temp2);
+            *start = NULL;
+        }
+        else
+        {
+            (temp2->prev)->next = temp1;
+            temp1->prev = temp2->prev;
+            free(temp2);
+            temp2 = NULL;
+        }
+        printf("\nNode succesfully deleted\n\n");
+    }
+}
```

Comment:
Actually, I prefer an `if-else` ladder because I personally think that it makes code more readable and understandable. But what you have done is also fine. If you think it's necessary to change I can make the suggested changes.

Reply:
I don't mind if you keep it that way, just wanted to point you to a [good programming practice](https://deviq.com/design-patterns/guard-clause). If you understand the reasoning behind them, it would become clear why it's important. But alas, that's irrelevant to this PR.

Reply:
Thank you for the suggestions. Soon after making the changes, I'll get back to you. 🙂